data/transactions/logic/eval.go

```diff
@@ -4216,6 +4216,32 @@ func opAppGlobalGetEx(cx *EvalContext) error {
 	return nil
 }
 
+// ensureLocalDelta is used to get accountIdx that is usable in the LocalDeltas
+// of the EvalDelta. The input accountIdx is "tentative" - if it's longer than
+// txn.Accounts, then we may need to add the address into SharedAccounts, and
+// index into it.
+func (cx *EvalContext) ensureLocalDelta(accountIdx uint64, addr basics.Address) uint64 {
+	if accountIdx > uint64(len(cx.txn.Txn.Accounts)) {
+		// the returned accountIdx was just a signal that the account was
+		// not in txn, so we look in SharedAccounts, allocating space if needed.
+		found := false
+		for i, shared := range cx.txn.EvalDelta.SharedAccts {
+			if shared == addr {
+				found = true
+				accountIdx = uint64(len(cx.txn.Txn.Accounts) + 1 + i)
+			}
+		}
+		if !found {
+			cx.txn.EvalDelta.SharedAccts = append(cx.txn.EvalDelta.SharedAccts, addr)
+			accountIdx = uint64(len(cx.txn.Txn.Accounts) + len(cx.txn.EvalDelta.SharedAccts))
```

Comment:
len(Accounts) + len(SharedAccts) - 1 (for 0 indexed) + 1 (to account for the signal index)

Reply:
In a sense, these are 1-based, because 0 is used to mean txn.Sender.

No room is allocated to the "signal index". When len(Accounts)+1 is returned from that other code, ironically, that value may actually end up being fine.  The first account in SharedAccounts uses that value. But, it's still a fine signal to the caller that we have a case that, if the caller intends to mutate, it has to find the Account (yet it might end up using the same value!)